addons/mail/static/src/models/message/message.js

```diff
@@ -442,6 +442,33 @@ function factory(dependencies) {
                 this.originThread.isModeratedByCurrentPartner
             );
         }
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsSubjectEqualThreadName() {
+            if (!this.subject || !this.originThreadName) {
+                return false;
+            }
+
+            const threadName = this.originThreadName.toLowerCase().trim();
+            let prefix = ['re:', 'fw:', 'fwd:'];
+            let startingSubject = this.subject.toLowerCase();
+            let cleanedSubject;
+            while (startingSubject !== cleanedSubject) {
+                cleanedSubject = startingSubject;
+                for (const pref of prefix) {
+                    if (startingSubject.startsWith(pref)) {
+                        startingSubject = startingSubject.replace(pref, '').trim();
+                        prefix = prefix.filter((pr) => pr !== pref);
```

Comment:
This line shouldn't be needed now, the `break` makes sure that once we have found one prefix, we start the loop over.
If you filter the prefix, multiple "re: re:" will not be caught correctly. That also means we need a new test, with 2x the same prefix, to cover this issue.